### ISSUE TITLE
Return separately values from the customized  Keras DataGenerator

### ISSUE THREAD

**Issue by NassimaD** [keras-team/keras#14254]
I created my custom data generator to return the fineNames IDs for each batch to use them for another purpose.
But when I want to get only the Images and labels for training (to fit them in the model.fit_generator). How can I do it. 

`
   class DataGenerator(keras.utils.Sequence):

    def __init__(self, extract, directory, list_IDs, labels, batch_size, dim, n_channels,
                 n_classes=10, shuffle=True):
        self.extract = extract
        self.directory = directory
        self.dim = dim
        self.batch_size = batch_size
        self.labels = labels
        self.list_IDs = list_IDs
        self.n_channels = n_channels
        self.n_classes = n_classes
        self.shuffle = shuffle
        self.on_epoch_end()

    def __len__(self):
        return int(len(self.list_IDs) / self.batch_size)+1

    def __getitem__(self, index):
        # Generate indexes of the batch
        indexes = self.indexes[index*self.batch_size:(index+1)*self.batch_size]

        # Find list of IDs
        list_IDs_temp = [self.list_IDs[k] for k in indexes]

        # Generate data
        X, y = self.__data_generation(list_IDs_temp)

        return [X, y], list_IDs_temp

    def on_epoch_end(self):
        self.indexes = np.arange(len(self.list_IDs))
        if self.shuffle == True:
            np.random.shuffle(self.indexes)

    def __data_generation(self, list_IDs_temp):
        # Initialization
        X = np.empty((len(list_IDs_temp), *self.dim, self.n_channels), dtype=np.float32)
        y = np.empty((len(list_IDs_temp)), dtype=int)

        # Generate data
        for i, ID in enumerate(list_IDs_temp):
            # Store sample
            X[i,] = load_image(os.path.join(self.directory, ID))
            # Store class
            if not self.extract:
                y[i] = self.labels[ID]
        if (self.extract):
            return X, []
        else:
            return X, keras.utils.to_categorical(y, num_classes=self.n_classes)`

I called the Generator on training and validation data and I want to get filenames and images and their labels separately withour iterating on the generator to fit them to the Model like: 

`
    validation_generator = DataGenerator(extract=False, directory=valid_dir, list_IDs=partition['validation'], labels=labels,
                                       shuffle=True, **params)

    train_generator = DataGenerator(extract=False, directory=train_dir, list_IDs=partition['train'], labels=labels,
                                       shuffle=True, **params)
    model_info = model.fit_generator(
        generator=train_generator,
        steps_per_epoch=steps_per_epoc,
        epochs=Num_epoach,
        validation_data=validation_generator,
        validation_steps=step_size_valid
 
    )` 
this code does not work because the generator return 3 values (IDs, Images,Labels) and I want to fit the model only  (IDs, Images) and get separately IDs

**Comment by sushreebarsa** [comment]
@NassimaD  Moving this issue to closed status as there has been no recent activity, in case you still face the error please create a new issue.Thanks!
